docs/source/notes/threading_cpu_inference.rst

```diff
@@ -0,0 +1,88 @@
+.. _threading-cpu-inference:
+
+Threading and CPU inference
+===========================
+
+PyTorch allows using multiple CPU threads during model inference. The following
+figure shows different levels of parallelism one would find in a typical
+application:
+
+.. image:: threading_cpu_inference.png
+   :width: 75%
+
+One or more inference threads execute model's forward pass on the given inputs.
+Each inference thread invokes a JIT interpreter that executes the ops
+of a model inline, one by one. A model can utilize a ``fork`` TorchScript
+primitive to launch some of the ops in parallel, as an asynchronous task, and
+synchronize with them later, using a future returned from ``fork``.
+
+PyTorch uses a single thread pool for the inter-op parallelism, this thread pool
+is shared by all inference tasks within the application process.
+
+Each operator can, in turn, use multiple threads to execute work. This can
+be useful in many cases, including element-wise ops on large tensors,
+convolutions, GEMMs, embedding lookups and others.
+
+PyTorch provides a set of parallel primitives (in ``ATen/Parallel.h``), such as
+``at::parallel_for``, that can be used to implement parallel ops.
+In addition to the functions in ``ATen/Parallel.h``, parallel ops can also
+utilize external libraries, such as MKL and MKL-DNN, to execute work.
+``ATen/Parallel.h``, MKL, MKL-DNN and other libraries typically use parallelization
+libraries (e.g. OpenMP or TBB) to implement multithreading. Intra-op thread
+management is delegated to the parallelization library and is separate from the
+inter-op thread pool.
+
+
+Build options
+-------------
+
+PyTorch supports OpenMP and TBB implementations of intra-op parallelism specified
+with the build settings:
+
```

Comment:
Add a pragraph stating why you would want to use multiple libraries and what trade-offs they might have -- or why the user should be interested in these build options / build settings.